kernel/aster-nix/src/fs/ext2/inode.rs

```diff
@@ -1496,12 +1552,14 @@ impl InodeImpl {
     pub fn inc_hard_links(&self) {
         let mut inner = self.0.write();
         inner.desc.hard_links += 1;
+        inner.desc.ctime = inner.current_time();
     }
 
     pub fn dec_hard_links(&self) {
         let mut inner = self.0.write();
         debug_assert!(inner.desc.hard_links > 0);
         inner.desc.hard_links -= 1;
+        inner.desc.ctime = inner.current_time();
```

Comment:
I don't think changing `ctime` in `dec_hard_links` method is a good idea.

```rust
impl Inode {
    pub fn rmdir(&self, name: &str) -> Result<()> {
        ...

        let mut dir_inner = dir_inner.upgrade();
        dir_inner.dec_hard_links();
        dir_inner.dec_hard_links(); // For "."
    }
}
```

For example, `Inode::rmdir` has updated the ctime of the deleted dir inode twice. This is obviously inefficient.

Another problem is that `Inode::rmdir` should have updated the ctime and mtime of the parent dir inode itself.